projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -116,20 +116,22 @@ public function get_preferred_view( $screen, $fallback_global_preference = true
 	 */
 	public function add_plugins_menu() {
 		global $submenu;
-		if (
-			isset( $submenu['plugins.php'] )
-			/**
-			 * Whether to enable the marketplace feature entrypoint.
-			 * This filter is specific to WPCOM, that's why there is no
-			 * need to use `jetpack_` prefix.
-			 *
-			 * @use add_filter( 'wpcom_marketplace_enabled', '__return_true' );
-			 * @module masterbar
-			 * @since 10.3
-			 * @param bool $wpcom_marketplace_enabled Load the WordPress.com Marketplace feature. Default to false.
-			 */
-			&& apply_filters( 'wpcom_marketplace_enabled', false )
-		) {
+
+		if ( ! isset( $submenu['plugins.php'] ) ) {
+			return;
+		}
+
+		/**
+		 * Whether to enable the marketplace feature entrypoint.
+		 * This filter is specific to WPCOM, that's why there is no
+		 * need to use `jetpack_` prefix.
+		 *
+		 * @use add_filter( 'wpcom_marketplace_enabled', '__return_true' );
+		 * @module masterbar
+		 * @since 10.3
+		 * @param bool $wpcom_marketplace_enabled Load the WordPress.com Marketplace feature. Default to false.
+		 */
+		if ( apply_filters( 'wpcom_marketplace_enabled', false ) ) {
```

Comment:
Maybe @obenland thought that because the action is prefixed with `wpcom_`, it only needs to run in the `wpcom` codebase?

However, this filter is triggered by Jetpack and (I guess) intended for targeting WP.com Simple and Atomic sites, which is why we cannot remove from this file.

Reply:
We're in `class-atomic-admin-menu.php`, which should only Ron un Atomic, no?

Reply:
> The filter comment says This filter is specific to WPCOM.

It think it wanted to mean WP.com sites in general (both Simple and Atomic). Am I right @cpapazoglou?

Reply:
exactly. Perhaps we can change
> This filter is specific to WPCOM

to
> This filter is specific to WP.com sites

Reply:
Does `class-atomic-admin-menu.php` get loaded on wpcom?

Reply:
No, but the `wpcom_marketplace_enabled` filter can be used anywhere. The comment is about the filter not the usage of it here, right?

Anyway, we can safely remove this now since we have launched the Marketplace and then we can also remove it from https://github.com/Automattic/wpcomsh/blob/51a665b80c90741041fedba8b040bdbb24269953/feature-plugins/masterbar.php#L269

Reply:
I see your point. The filter was added so that we could launch the Marketplace through wpcomsh regardless Jetpack release.

Reply:
I was playing around a bit to see the behavior if we can remove the `add_plugins_menu` function completely. I was also checking the behavior if we remove changes added for the Marketplace within the function. 

- Removing the `add_plugins_menu` function completely will fall back to [parent class menu](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L364) `add_plugins_menu` function, which will remove the plugin install & plugin editor menu items from the menu. The plugins menu will force users to use the Calypso screen. I believe that's the plugins menu we want to keep for free Atomic sites. 
<img width="289" alt="Screenshot 2022-02-16 at 11 44 32 AM" src="https://cloudup.com/cfITrXCK5vS+">

- If we keep the empty `add_plugins_menu` function body and remove all changes added for the Marketplace then the plugins menu will fall back to WordPress Core Behavior (IIRC). I believe that's the plugins menu we want to keep for the Premium Atomic site. 
<img width="289" alt="Screenshot 2022-02-16 at 11 44 32 AM" src="https://user-images.githubusercontent.com/86406124/154207128-74f1b142-59e3-4ad0-80dd-713105ec052f.png">

It is certain that without knowing the type of Atomic site (plan check), we can't really impose menu behavior on the user from this class. 

Here is my suggestion, We keep WordPress core behavior in this file by removing all changes added for the Marketplace and keeping the `add_plugins_menu` function body empty. At the same time, We make changes in WPCOMSH to impose the Free Atomic plugins menu. WPCOMSH will remove the existing plugins menu and new plugins menu items to force calypso screens.

Reply:
> It is certain that without knowing the type of Atomic site (eligibility check), we can't really impose menu behavior on the user from this class.

`Atomic_Plan_Manager` is available here, so we could perform some plan eligibility checks within this class. Once we move to a feature eligibility checks model, we would need to refactor the logic, but in the meantime I think it's fine.

Reply:
To summarize, I think this is the behavior we want for the Plugins menu:

Site | Parent menu | Submenus
--- | --- | ---
Simple | `https://wordpress.com/plugins/:site` | No
Atomic (Business, eCommerce, Marketplace addon) | `https://wordpress.com/plugins/:site` | <ul><li>Add new: `https://wordpress.com/plugins/:site`</li><li>Installed Plugins: `wp-admin/plugins.php`</li><li>Plugin File Editor: `wp-admin/plugin-editor.php`</li></ul>
Atomic (Free, Premium, no marketplace addons) | `https://wordpress.com/plugins/:site` | No
Jetpack | `https://wordpress.com/plugins/:site` | No

Reply:
Thanks @mmtr for summarising it. I will refactor it. 

Are you sure about giving access to WP admin screens for plugin management when user has purchased only `Marketplace addon`?  My understanding is that, we want to control user's ability to perform actions on site when site is Atomic with lower plan and marketplace addon.

Reply:
@gavande1 you're right, I mistakenly assumed that was the behavior right now, but I just tested on my Atomic site which only has a marketplace addon and confirmed that the menu gets the same behavior as sites on Free/Premium plans.

Updated the table above to reflect this.

Reply:
> Atomic_Plan_Manager is available here, so we could perform some plan eligibility checks within this class. Once we move to a feature eligibility checks model, we would need to refactor the logic, but in the meantime, I think it's fine.

I think we need a better way to ask if the site is Free Atomic. IMHO, Using`Atomic_Plan_Manager` within Jetpack seems odd to me. However, I've made the changes for now. As @mmtr mentioned, we can iterate over it again once we have a better way available.

Reply:
> Using `Atomic_Plan_Manager` within Jetpack seems odd to me. 

Yeah, I realize that using code here that isn't directly available in the Jetpack repo can be weird.

However, we already do that for Simple sites, where we use functions and classes within Jetpack that are only available in `wpcom` ([example 1](https://github.com/Automattic/jetpack/blob/2e13b50883d09cfc6c29ec9d987c06a3a84c7b86/projects/plugins/jetpack/modules/masterbar/admin-menu/class-dashboard-switcher-tracking.php#L99), [example 2](https://github.com/Automattic/jetpack/blob/2e13b50883d09cfc6c29ec9d987c06a3a84c7b86/projects/plugins/jetpack/modules/masterbar/admin-menu/class-p2-admin-menu.php#L79-L80), [example 3](https://github.com/Automattic/jetpack/blob/2e13b50883d09cfc6c29ec9d987c06a3a84c7b86/projects/plugins/jetpack/modules/masterbar/admin-menu/load.php#L67)), so this wouldn't be any different.